python/pyspark/mllib/clustering.py

```diff
@@ -346,7 +346,7 @@ def train(cls, rdd, k, convergenceTol=1e-3, maxIterations=100, seed=None, initia
             if initialModel.k != k:
                 raise Exception("Mismatched cluster count, initialModel.k = %s, however k = %s"
                                 % (initialModel.k, k))
-            initialModelWeights = initialModel.weights
+            initialModelWeights = [w for w in initialModel.weights]
```

Comment:
Instead of doing this, I suggest you explicitly call `list(initialModel.weights)`, it is easier to understand.